app/scripts/lib/multichain-api/wallet-requestPermissions.test.js

```diff
@@ -0,0 +1,369 @@
+      const { handler, requestPermissionsForOrigin, response } =
+        createMockedHandler();
+      requestPermissionsForOrigin.mockResolvedValue([
+        {
+          otherPermission: {
+            id: '3',
+            parentCapability: 'otherPermission',
+            caveats: [
+              {
+                value: {
+                  foo: 'bar',
+                },
+              },
+            ],
+          },
+        },
+      ]);
+
+      await handler(baseRequest);
+      expect(response.result).toStrictEqual([
+        {
+          id: '3',
+          parentCapability: 'otherPermission',
+          caveats: [
+            {
+              value: {
+                foo: 'bar',
+              },
+            },
+          ],
+        },
+      ]);
+    });
+
+    it('gets permission for the origin', async () => {
+      const { handler, getPermissionsForOrigin } = createMockedHandler();
+
+      await handler(baseRequest);
+      expect(getPermissionsForOrigin).toHaveBeenCalledWith('http://test.com');
+    });
+
+    describe('CAIP-25 permission does not exist', () => {
+      it('grants a new CAIP-25 endowment with an optional scope for the current chain', async () => {
+        const { handler, getPermissionsForOrigin, grantPermissions } =
+          createMockedHandler();
+        getPermissionsForOrigin.mockReturnValue(Object.freeze({}));
+
+        await handler(baseRequest);
+        expect(grantPermissions).toHaveBeenCalledWith({
+          subject: {
+            origin: 'http://test.com',
+          },
+          approvedPermissions: {
+            [Caip25EndowmentPermissionName]: {
+              caveats: [
+                {
+                  type: Caip25CaveatType,
+                  value: {
+                    requiredScopes: {},
+                    optionalScopes: {
+                      'eip155:1': {
+                        methods: [],
+                        notifications: [],
+                        accounts: ['eip155:1:0xdead', 'eip155:1:0xbeef'],
+                      },
+                    },
+                  },
+                },
+              ],
+            },
+          },
+        });
+      });
+
+      it('returns the granted permissions with the CAIP-25 endowment transformed into eth_accounts', async () => {
+        const { handler, getPermissionsForOrigin, response } =
+          createMockedHandler();
+        getPermissionsForOrigin.mockReturnValue(Object.freeze({}));
+
+        await handler(baseRequest);
+        expect(response.result).toStrictEqual([
+          {
+            id: 'new',
+            parentCapability: 'eth_accounts',
+            caveats: [
+              {
+                value: ['0xdead', '0xbeef'],
+              },
+            ],
+          },
+        ]);
+      });
+    });
+
+    describe('CAIP-25 permission does exist', () => {
+      it('updates the existing CAIP-25 endowment with an optional scope for the current chain', async () => {
```

Comment:
might be worth adding a note for legibility that an account is added to an existing chain scope

Reply:
split this into two scenarios

https://github.com/MetaMask/metamask-extension/pull/26054/commits/07b76eda78ad733644299dbd23289c166cfd715c